vllm/v1/core/block_pool.py

```diff
@@ -36,7 +36,8 @@ def __init__(
         enable_caching: bool,
         enable_kv_cache_events: bool = False,
     ):
-        assert isinstance(num_gpu_blocks, int) and num_gpu_blocks > 0
+        # num_gpu_blocks can be 0 for attention free models
+        assert isinstance(num_gpu_blocks, int)
```

Comment:
I think we can always have at least 1 gpu block so that we don't need to handle null block.